docs/data/data-grid/filtering-recipes/FilterPresetsPanel.tsx

```diff
@@ -0,0 +1,369 @@
+import ListItem from '@mui/material/ListItem';
+import ListItemButton from '@mui/material/ListItemButton';
+import ListItemText from '@mui/material/ListItemText';
+import Typography from '@mui/material/Typography';
+import Alert from '@mui/material/Alert';
+import Stack from '@mui/material/Stack';
+import Tooltip from '@mui/material/Tooltip';
+import Divider from '@mui/material/Divider';
+import SaveIcon from '@mui/icons-material/Save';
+import DeleteIcon from '@mui/icons-material/Delete';
+import AddIcon from '@mui/icons-material/Add';
+
+interface FilterPreset {
+  id: string;
+  name: string;
+  filterModel: GridFilterModel;
+  createdAt: string;
+}
+
+interface FilterPresetStore {
+  presets: FilterPreset[];
+  activePresetId: string | null;
+}
+
+const STORAGE_KEY = 'dataGridFilterPresets';
+const EMPTY_STORE: FilterPresetStore = { presets: [], activePresetId: null };
+
+const createPresetsStore = () => {
+  let listeners: Array<() => void> = [];
+
+  return {
+    subscribe: (callback: () => void) => {
+      listeners.push(callback);
+      return () => {
+        listeners = listeners.filter((listener) => listener !== callback);
+      };
+    },
+    getSnapshot: () => {
+      try {
+        const saved = localStorage.getItem(STORAGE_KEY);
+        return saved || JSON.stringify(EMPTY_STORE);
+      } catch {
+        return JSON.stringify(EMPTY_STORE);
+      }
+    },
+    getServerSnapshot: () => {
+      return JSON.stringify(EMPTY_STORE);
+    },
+    update: (store: FilterPresetStore) => {
+      try {
+        localStorage.setItem(STORAGE_KEY, JSON.stringify(store));
+      } catch {
+        // Silently fail if localStorage is not available
+      }
+      listeners.forEach((listener) => listener());
+    },
+  };
+};
+
+const usePersistedPresets = () => {
+  const [presetsStore] = React.useState(createPresetsStore);
+
+  const storeString = React.useSyncExternalStore(
+    presetsStore.subscribe,
+    presetsStore.getSnapshot,
+    presetsStore.getServerSnapshot,
+  );
+
+  const store = React.useMemo(() => {
+    try {
+      return JSON.parse(storeString) as FilterPresetStore;
+    } catch {
+      return EMPTY_STORE;
+    }
+  }, [storeString]);
+
+  return React.useMemo(
+    () => [store, presetsStore.update] as const,
+    [store, presetsStore.update],
+  );
+};
+
+type GridFilterPanelProps = React.ComponentProps<typeof GridFilterPanel>;
+
+function CustomFilterPanel(props: GridFilterPanelProps) {
+  const apiRef = useGridApiContext();
+  const [store, setStore] = usePersistedPresets();
+  const [createFilterDialogOpen, setCreateFilterDialogOpen] = React.useState(false);
+  const [createFilterName, setCreateFilterName] = React.useState('');
+
+  const getCurrentFilterModel = () => apiRef.current.state.filter.filterModel;
+  const hasActiveFilters = () => getCurrentFilterModel().items.length > 0;
+
+  const handleSavePreset = () => {
+    if (!store.activePresetId) {
+      return;
+    }
```

Comment:
Nit: This condition is unlikely to be met, since the button will be disabled on this condition anyways.
```suggestion
```

Reply:
I think the function should be independent from UI to avoid error when `activePresetId` is not present (somehow).

Reply:
But, better to remove it since without `activePresetId`, the code still function.